rfc/system/3749-sane-overridable-defaults.md

```diff
@@ -0,0 +1,208 @@
+        - project: flytesnacks
+          domain: development
+          overrides:
+               workflow_execution_config:
+                    labels:
+                        my-key: "my-value"
+                    security_context:
+                        core_identity:
+                            iam_role: "my_iam_role"
+        ...                             
+
+```
+This can work the same way cluster resource templates are mounted as ConfigMaps in the FlyteAdmin deployment but are [fully customizable](https://github.com/flyteorg/flyte/blob/5ab8cb887a5a7051070cb93fca603ed1f22f2f74/charts/flyte-core/values.yaml#L752-L810).
+
+This does require restarting the FlyteAdmin deployment for invididual overrides changes but allows for declaratively defining the complete set of overrides in a reproducible manner. See more discusison on the trade-offs [here](https://github.com/flyteorg/flyte/discussions/3731#discussioncomment-6053743).
+
+#### FlyteAdmin changes
+
+##### Deprecated DB: Get Overrides
+Getting overrides at a super specific level, for example at the workflow level, should use the same fallback logic currently applied at execution time to determine all applicable overrides across all matchable resource types (task resources, workflow execution config, etc).
+
+That is, the workflow-level document should include all applicable workflow overrides, any project overrides and any domain overrides in one document with inherited overrides marked as such.
+
+##### Single ConfigMap
+When the FlyteAdmin application starts, it can load the overridable defaults configmap and create an in-memory the mapping of per-project, per-domain and per-workflow overrides.
+
+At execution time when overrides need to be applied or when a reader calls the `GetOverrides` API directly, the mapping can be queried for the applicable per-project, per-domain and per-workflow overrides, much like the logic currently applied at execution time.
+
+##### Updating overrides
+When FlyteAdmin is configured to use the database as the source of overrides, the existing flytectl CLI methods can be used to set matchable overrides from the command line.
+
+When FlyteAdmin is configured to use the configmap as the source, any attempts to use the MatchableResource write API should fail (that is calls to `Update*Attributes` and `Delete*Attributes` endpoints)
+
+### Migration Strategy
+
+Eventually the single, entire document stored in the ConfigMap should be the source of truth instead of the current resources table in the database.
+
+This allows us to do a finite number of in-memory queries at execution time to fetch the entire list of realized overrides for the global, domain, project and workflow level for all overridable attributes
+
+Because this RFC proposes one source of truth for overrides, either the deprecated `db` or `configmap` types, Flyteadmin could ship with an additional endpoint to spit out a complete ConfigMap that captures the full set of all overrides
```

Comment:
I personally wouldn't use flytectl to view the configmap but always kubectl so I'd say such an endpoint might not be needed if the configmap is the single source of truth and should match the in memory reprentation of flytemadmin.